File: exam-conductor/archiveDCR/libs/exampen-proto/typescript/src/ai.ts
/** AI pipeline types: recognition results, confidence, step breakdowns. */

import type { AISourceType } from "./enums";

/** AI recognition result for a single question. */
export interface QuestionResult {
  question_id: string;
  recognized_text: string;
  confidence: number;
  step_breakdown?: string[];
}

/** NATS event: AI recognition complete for a student's exam. */
export interface AIResultEvent {
  event_id: string;
  event_type: "ai.result";
  event_version: "1.0.0";
  occurred_at: string;
  exam_id: string;
  student_id: string;
  model_version: string;
  source_type?: AISourceType;
  question_results: QuestionResult[];
}

/** Student-facing answer detail with AI analysis. */
export interface AnswerInsight {
  question_id: string;
  answer_image_uri: string;
  recognized_text: string;
  confidence: number;
  step_breakdown?: string[];
  feedback?: string;
}

File: exam-conductor/archiveDCR/libs/exampen-proto/python/exampen_proto/ai.py
"""AI pipeline models: recognition results, confidence, step breakdowns."""

from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel

from .enums import AISourceType


class QuestionResult(BaseModel):
    """AI recognition result for a single question."""

    question_id: str
    recognized_text: str
    confidence: float
    step_breakdown: Optional[list[str]] = None


class AIResultEvent(BaseModel):
    """NATS event: AI recognition complete for a student's exam."""

    event_id: str
    event_type: str = "ai.result"
    event_version: str = "1.0.0"
    occurred_at: datetime
    exam_id: UUID
    student_id: str
    model_version: str
    source_type: Optional[AISourceType] = None
    question_results: list[QuestionResult]


class AnswerInsight(BaseModel):
    """Student-facing answer detail with AI analysis."""

    question_id: str
    answer_image_uri: str
    recognized_text: str
    confidence: float
    step_breakdown: Optional[list[str]] = None
    feedback: Optional[str] = None
